File: Recursion/lecture 2/Fraz_recursion_2.java
import java.io.*;
import java.util.*;

class Main {
    static int factorial(int n) {
        System.out.println(n);
        if (n == 1)
            return 1;
        return n * factorial(n - 1);
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        if (n < 0) {
            System.out.println("Error");
        }
        System.out.println(factorial(n));
    }
}

File: Recursion/lecture 2/Fraz_recursion_2.py
# Factorial of a number using recursion

def factorial(n):
   print(n)
   if n == 0:
       return n
   else:
       return n*factorial(n-1)


n=input();
if int(n) < 0:
   print("Error")
else:
   print(factorial(int(n)))